### ISSUE TITLE
Table header elements are not given role button

### ISSUE THREAD

**Issue by ksarkerhy** [IgniteUI/ignite-ui#2247]
## Description  
Describe the issue.
We are consumers of iggrid, and it has been used for various features in our application. During a recent evaluation by a third-party accessibility vendor, it came to our attention that there is no button role or <button> tag used in table header elements of igGrid. Even though they can be used as buttons to sort the columns. This is necessary according to the WCAG 2.0 Level A guideline 4.1.2 - Name, Role, Value.

 * ignite-ui version: Any
 * browser: Any

## Steps to reproduce  

1. Go to the [sample](https://jsfiddle.net/gh/get/jquery/1.9.1/IgniteUI/help-samples/tree/master/23.1/EN/grid/sorting-local/fiddle).
2. Use NVDA or another accessibility tool.
3. Navigate to table header elements Employee ID, First Name, etc.

## Result  
The screen reader is reading "Employee ID column header Sort column row 1 Employee ID column 1"

## Expected result  
The screen reader should read the elements as button.
Using role="button" or using proper <button> tag for the elements.

## Attachments  
![iggrid2](https://github.com/IgniteUI/ignite-ui/assets/160117217/386c2368-79b0-4f17-bb65-80f7df8f9b5e)

**Comment by ChronosSF** [comment]
@ksarkerhy , I don't agree that the headers should have `role="button"` only and unfortunately multiple roles, while technically supported are not supported by e.g. NVDA which only uses the first valid value. If you manually change a TH to have role button you will see that a lot of the important information that it usually reads for column headers is lost as the default readout for buttons has nothing to do with what you'd want read for column headers. 

What we can do is change the label a bit so it prompts that clicking sorts the column.

**Comment by ChronosSF** [comment]
Double checking this, it seems that we already provide a title that prompts the user that the column is sortable. As we won't be making the role to button as that regresses the readout on TH elements severely, I think the title is more or less the best we can do in this case.
